secboot/secboot_sb.go

```diff
@@ -161,6 +161,22 @@ func UnlockVolumeUsingSealedKeyIfEncrypted(disk disks.Disk, name string, sealedE
 		return res, fmt.Errorf("internal error: cannot build an auth requestor: %v", err)
 	}
 
+	if loadedKey.SealedKeyV1 != nil {
```

Comment:
this feels like it's a leaky abstraction which only muddies the whole picture, could we do something more explicit, eg:

```
// legacy fde-hooks keys which did not carry any metadata
var legacyV1Key []byte
// more modern keys with metadata
var keys []*sb.KeyData

// isLegacyV1Key - simply checks the header

if fdeHasRevealKey() && isLegacyV1Key(sealedEncryptioKeyFile) {
	legacyV1Key, err = os.ReadFile()
	...
} else {
	if err := readKeyFile(sealedEncryptionKeyFile, loadedKey); err != nil {
		return res, err
	}
}

...
if len(legacyV1KeyFile) != 0 {

}
...
// continue with ActivateVolumeWithKeyData
```

and then simply drop the loader thingy